### PR TITLE
fix: Archiving logs loses big amount of data

### LINKED ISSUE #1080: Archiving logs loses big amount of data
### Environment

- Smapp version: 0.2.10
- Node version: v0.2.20-beta.0
- OS: macOS Ventura Version 13.0.1 + Windows 10 Home

### Describe the bug

Once in a while, the _spacemesh-log_ file gets compressed/archived. This process causes loosing data - in the uncompressed files, there are significant shortages of data (several hours of logs missing). 

### Steps to reproduce

1. Open Smapp and note the start time
2. Leave the Smapp running with no breaks for a long period of time (a day or two).
3. Check the logs
4. Compare what has been saved in the .gz archive and app-log.
5. Compare the first timestamp in the oldest _spacemesh-log_ file and the actual time when the app was started and the logs should have been saved. 

### Expected behavior

All the logs should be saved and safely stored in the Spacemesh directory. The compressing process shouldn’t cut any data.

### Actual behavior

Several hours of logs disappear once the _spacemesh-log_ file gets archived. 

Screenshot 2023-01-09 at 08 43 31
Screenshot 2023-01-09 at 08 27 32

Screenshot 2023-01-09 at 08 29 55

### PR DESCRIPTION
**Pull request by maparr**
Try to close #1080 
Screenshot 2023-01-27 at 13 42 25

30s-2 min gap , because of check, at least it will resolve our issue with hours lost data

### CODE REVIEW

**Comment by maparr** [comment]
Hi @monikasmolarek , when u will have time may u check

**Comment by brusherru** [comment]
@maparr `logrotator` lib is using `copytruncate` under the hood
Just replace it with https://www.npmjs.com/package/logrotate-stream and pipe stdout&stderr right there.
I believe it should work :)

@monikasmolarek so no need to check it yet ;)
Tweaking properties may improve our chances, but they won't completely eliminate the problem

**Comment by maparr** [comment]
@brusherru checked this lib, a problem with rotation, if rotation files exceeded the max level, logs won't be rotated and will be streamed to the file with no restriction. Need to find another lib or fork it 
Screenshot 2023-01-27 at 21 43 09

**Comment by maparr** [comment]
works fine, with 5k size rotation 
Screenshot 2023-01-27 at 22 27 16
cc: @brusherru

File: types/logrotate-stream.d.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// Type definitions for logrotate-stream 0.2.5
```
Comment (brusherru) [inline types/logrotate-stream.d.ts]: Why it is here instead of package.json `yarn add -D @types/logrotate-stream`? :)

**Review by brusherru — CHANGES_REQUESTED** [review]
1. Do not move `fonts.d.ts` and `images.d.ts` to the "global" scope. It is related only to GUI — so let's keep it there.
2. Don't repeat the code ;)

**Comment by monikasmolarek** [comment]
Tested on Windows and worked pretty well; however, on Mac, the file that reached already 104Mb didn’t get archived. Continued to be used/updated. I will re-test it cause I fumbled a bit in the log file to speed up the whole process, so it might be my fault.

File: desktop/NodeManager.ts [hunk 2]
```diff
@@ -392,16 +388,13 @@ class NodeManager extends AbstractManager {
...
     const logFilePath = getNodeLogsPath(this.genesisID);
 
-    rotator.register(logFilePath, {
-      schedule: '30m',
-      size: '500m',
-      count: 1, // number of old logs files that ll be saved and compressed
+    const logFileStream = rotator({
+      file: logFilePath,
+      size: '100m',
+      keep: 5,
```
Comment (monikasmolarek) [inline desktop/NodeManager.ts:394]: I trust it will be enough, the decision is yours Guys :)

**Review by monikasmolarek — APPROVED** [review]
looks ok :) if no issues with smapp, network etc. it works fine.

**Review by brusherru — APPROVED** [review]

**Pull request merged**